src/WinSparkleDotNet/WinSparkle.cs

```diff
@@ -0,0 +1,557 @@
+                SetAutomaticCheckForUpdates(value ? 1 : 0);
+            }
+        }
+
+#region DLLWrapping
+        /// <summary>   
+        ///     <para>
+        ///         Starts WinSparkle.
+        ///     </para>
+        ///     <para>
+        ///         If WinSparkle is configured to check for updates on startup, proceeds
+        ///         to perform the check. You should only call this function when your app
+        ///         is initialized and shows its main window.
+        ///     </para>
+        /// </summary>
+        ///
+        /// <remarks>   This call doesn't block and returns almost immediately. If an
+        ///          update is available, the respective UI is shown later from a separate
+        ///          thread. </remarks>
+        /// 
+        /// <see cref="Cleanup"/>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_init", CallingConvention = CallingConvention.Cdecl)]
+        public static extern void Initialize();
+
+        /// <summary>  Cleans up after WinSparkle.
+        ///    Should be called by the app when it's shutting down. Cancels any
+        ///    pending Sparkle operations and shuts down its helper threads. </summary>
+        ///
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_cleanup", CallingConvention = CallingConvention.Cdecl)]
+        public static extern void Cleanup();
+
+        /// <summary>   
+        ///     <para>
+        ///          Sets UI language from its ISO code.
+        ///     </para>
+        /// </summary>
+        ///
+        /// <remarks>  This function must be called before <see cref="Initialize"/> </remarks>
+        ///
+        /// <param name="lang">   ISO 639 language code with an optional ISO 3116 country
+        ///                       code, e.g. "fr", "pt-PT", "pt-BR" or "pt_BR"
+        ///  </param>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_set_lang", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+        public static extern void SetLanguage(
+            [param: MarshalAs(UnmanagedType.AnsiBStr)] string lang
+            );
+
+        /// <summary>   
+        ///     <para>
+        ///         Sets URL for the app's appcast.
+        ///     </para>
+        ///     <para>
+        ///         If this function isn't called by the app, the URL is obtained from
+        ///         Windows resource named "FeedURL" of type "APPCAST".
+        ///     </para>
+        /// </summary>
+        ///
+        /// <remarks>  Can only be called @em before the first call to <see cref="Initialize"/> </remarks>
+        ///
+        /// <param name="url">   URL of the appcast. </param>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_set_appcast_url", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+        public static extern void SetAppcastUrl(
+            [param: MarshalAs(UnmanagedType.AnsiBStr)] string url
+            );
```

Comment:
Why do you marshal as `UnmanagedType.AnsiBStr` if the WinSparkle API is declared as follows?
```
WIN_SPARKLE_API void __cdecl win_sparkle_set_appcast_url(const char *url)
```
Using `UnmanagedType.LPStr` would be a better fit. An `UnmanagedType.AnsiBStr` is a [COM-style BSTR with a prefixed length and ANSI characters](https://msdn.microsoft.com/en-us/library/s9ts558h%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396).